homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,231 @@
+
+    key: str = CoverDeviceClass.BLIND.value
+    translation_key: str = CoverDeviceClass.BLIND.value
+
+
+SHADE_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.SHADE
+)
+BLIND_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.BLIND
+)
+CURTAIN_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.CURTAIN
+)
+VERTICAL_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.CURTAIN, icon=ICON_VERTICAL_BLIND
+)
+
+COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
+    MotionBlindType.HONEYCOMB.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.ROMAN.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.ROLLER.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.DOUBLE_ROLLER.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.VENETIAN.name: BLIND_ENTITY_DESCRIPTION,
+    MotionBlindType.VENETIAN_TILT_ONLY.name: BLIND_ENTITY_DESCRIPTION,
+    MotionBlindType.CURTAIN.name: CURTAIN_ENTITY_DESCRIPTION,
+    MotionBlindType.VERTICAL.name: VERTICAL_ENTITY_DESCRIPTION,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up blind based on a config entry."""
+
+    blind_class = BLIND_TYPE_TO_CLASS[entry.data[CONF_BLIND_TYPE].upper()]
+    device = hass.data[DOMAIN][entry.entry_id]
+    entity = blind_class(device, entry)
+
+    async_add_entities([entity])
+
+
+class MotionblindsBLECoverEntity(MotionblindsBLEEntity, CoverEntity):
+    """Representation of a blind."""
+
+    _attr_is_closed: bool | None = None
+    _attr_name = None
+
+    def __init__(self, device: MotionDevice, entry: ConfigEntry) -> None:
+        """Initialize the blind."""
+        _LOGGER.debug(
+            "(%s) Setting up %s cover entity (%s)",
+            entry.data[CONF_MAC_CODE],
+            entry.data[CONF_BLIND_TYPE].upper(),
+            BLIND_TYPE_TO_CLASS[entry.data[CONF_BLIND_TYPE].upper()].__name__,
+        )
+        super().__init__(
+            device, entry, COVER_TYPES[entry.data[CONF_BLIND_TYPE].upper()]
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register device callbacks."""
+        self.device.register_running_callback(self.async_update_running)
+        self.device.register_position_callback(self.async_update_position)
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop moving the blind."""
+        _LOGGER.debug("(%s) Stopping", self.entry.data[CONF_MAC_CODE])
+        await self.device.stop()
+
+    @callback
+    def async_update_running(
+        self, running_type: MotionRunningType | None, write_state: bool = True
+    ) -> None:
+        """Update whether the blind is running (opening/closing) or not."""
+        if running_type in [None, MotionRunningType.STILL, MotionRunningType.UNKNOWN]:
```

Comment:
```suggestion
        if running_type in {None, MotionRunningType.STILL, MotionRunningType.UNKNOWN}:
```

Probably better as a constant though